### DISCUSSION TITLE
bulk import bash script doesnt work

### DISCUSSION THREAD

**Discussion by BrandonG777** [mealie-recipes/mealie#439] [Q&A] [answered]
I get the following error whenever trying to run the script. Any ideas how I can make it work?

curl: (6) Could not resolve host: html><html

UPDATE: Removed special characters from my password and now I receive an Internal Server Error message. I have also exposed port 9000 and tried sending these request directly through the API server but I still get the same message.

2ND UPDATE: So despite receiving the Internal Server Errror messages my recipes did actually start importing! Further inspection also revealed this error was only coming from one or two of my recipe sites.

[recipes.txt](https://github.com/hay-kot/mealie/files/6549064/recipes.txt)

**Comment by BrandonG777** [comment, marked as answer]
Useful info that may help others...

1. Make sure your mealie_url doesn't have a trailing slash.
2. Make sure you password doesn't have any special characters, this may interfere with regex.
3. Make sure your URL list doesn't contain carriage returns.
4. Some sites may generate an error but don't assume (like I did) that your recipes aren't getting imported.

**Reply by tehniemer** [reply]
I've tried your suggestions and I'm still having problems. Every URL is returning `curl: (3) URL using bad/illegal format or missing URL` 

[recipes.txt](https://github.com/hay-kot/mealie/files/6605304/recipes.txt)

**Reply by BrandonG777** [reply]
Sounds like some sort of syntax problem but I really don't have anything else to suggest. The script does work it just has a lot of room for errors and isn't very helpful in resolving those errors. Also, keep in mind you don't have to run this on the same machine.

**Comment by sarudy89** [comment]
Hello @BrandonG777 I tried your bash script, and I am getting this error. Could you please direct me to what could be the issue? `URL rejected: Malformed input to a URL function` Thank  you! 
![Screenshot 2024-06-24 083951](https://github.com/mealie-recipes/mealie/assets/147134760/2a832191-2c32-4a45-a44c-bfdb503cdcfd)
